### PR TITLE
TTE-21

### PR DESCRIPTION
**Pull request by jzcodeveloper**
- Added custom hooks usePrevious and useInterval
- Active video is automatically played
- Previous active video gets paused

### CODE REVIEW

File: src/utils/app.constant.js [hunk 1]
```diff
@@ -1,2 +1,2 @@
-export const MAIN_API_URL = 'https://tiktok-educative-api.herokuapp.com/api/v1';
```
[the remark this answers is not in the document]
Comment (jzcodeveloper) [reply src/utils/app.constant.js:1]: My bad, I'll fix it

File: src/components/video-player/index.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+  const setVideoRef = async component => {
+    if (videoRef.current) return;
+
+    try {
+      videoRef.current = component;
+      await videoRef.current.loadAsync({ uri: source }, { shouldPlay: false });
+      setLoading(false);
+    } catch (error) {
+      console.log(error);
```
[the remark this answers is not in the document]
Comment (jzcodeveloper) [reply src/components/video-player/index.js:37]: Quiet high for slow connections

File: src/utils/hooks.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  const prevState = useRef();
+
+  useEffect(() => {
+    prevState.current = value;
+  }, [value]);
+
+  return prevState.current;
+};
+
+export const useInterval = (callback, delay) => {
```
[the remark this answers is not in the document]
Comment (jzcodeveloper) [reply src/utils/hooks.js:13]: Not using it right now, I though I could use in the VideoPlayer component but seems like it's not necessary

File: src/utils/app.constant.js [hunk 4]
```diff
@@ -1,2 +1,2 @@
...
+export const MAIN_API_URL = 'http://192.168.1.10:5000/api/v1';
```
[the remark this answers is not in the document]
Comment (jzcodeveloper) [reply src/utils/app.constant.js:1]: Excuse that, I'll consider using NODE_ENV to set this variable

File: src/utils/hooks.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (lyondeveloper) [reply src/utils/hooks.js]: if so, please remove it

File: src/utils/app.constant.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (lyondeveloper) [reply src/utils/app.constant.js]: the local IPs will change, leave it like that

File: src/utils/hooks.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (jzcodeveloper) [reply src/utils/hooks.js]: done

File: src/utils/app.constant.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (jzcodeveloper) [reply src/utils/app.constant.js]: this should be the remote url

[the remark this answers is not in the document]
Comment (lyondeveloper) [reply src/utils/app.constant.js]: yeah but I was meaning the ip number you put on, for running locally the api, but that changes

**Review by lyondeveloper — APPROVED** [review]

**Pull request merged**